plugin.video.srgssr_ch_replay/resources/lib/plugin.py

```diff
@@ -0,0 +1,362 @@
+        try:
+            userdata_path = xbmcvfs.translatePath(self.ADDON.getAddonInfo('profile')).decode('utf-8')
+        except AttributeError:
+            userdata_path = xbmcvfs.translatePath(self.ADDON.getAddonInfo('profile'))
+        
+        if not os.path.isdir(userdata_path):
+            os.mkdir(userdata_path)
+
+    def _create_api_clients(self) -> Tuple[SRGSSRVideoApiClient, SRGSSRSubtitlesApiClient]:
+        """Creates and returns the Video and Subtitles API clients"""
+        self._check_api_credentials_set("consumer_key", "consumer_secret")
+        video_client = SRGSSRVideoApiClient(
+            self.SRG_API_BASE_URL,
+            {"key": self.settings.consumer_key, "secret": self.settings.consumer_secret},
+            self,
+        )
+        
+        subs_client = None
+        if self.settings.enable_subtitles == "true":
+            self._check_api_credentials_set("consumer_key_subtitles", "consumer_secret_subtitles")
+            subs_client = SRGSSRSubtitlesApiClient(
+                self.SRG_API_BASE_URL,
+                {"key": self.settings.consumer_key_subtitles, "secret": self.settings.consumer_secret_subtitles},
+                self,
+            )
+
+        return (video_client, subs_client)
+
+    def _check_api_credentials_set(self, key_setting: str, secret_setting: str):
+        """Checks that Video or Subtitles API credentials are set, and open the settings if not"""
+        while getattr(self.settings, key_setting) == "" or getattr(self.settings, secret_setting) == "":
+            xbmcgui.Dialog().ok(self.tr(30099), self.tr(30098))
+            self.ADDON.openSettings()
+
+    def run(self):
+        """Plugin main method"""
+        self.logger.info("Starting SRGSSR plugin")
+        self.router(sys.argv[2][1:])    # passes the url params to the router
+        self.logger.info("End of SRGSSR plugin")
+
+    def router(self, paramstring: str):
+        """Routes to the different pages or actions of the plugin
+        :param paramstring: The URL parameters (after the ?)
+        """
+        params = dict(parse_qsl(paramstring))
+
+        mode = params.get("mode", "")
+        bu = params.get("channel")
+
+        if mode == "playEpisode":
+            episode_id = params.get("episodeId", "")
+            media_id = params.get("mediaId", "")
+            self.play_episode(bu, episode_id, media_id)
+        elif mode == "listEpisodes":
+            tvshow_id = params.get("tvShowId")
+            current_page = int(params.get("currentPage", 1))
+            number_of_episodes = int(params.get("numberOfEpisodes", 0))
+            next_page_id = params.get("nextPageId", "")
+            self.list_episodes(bu, tvshow_id, current_page, number_of_episodes, next_page_id)
+        elif mode == "listTvShowsByLetter":
+            letter = params.get("letter")
+            self.list_tv_shows(bu, letter)
+        elif mode == "searchTvShows":
+            self.search_tv_shows()
```

Comment:
search_tv_shows() is not implemented

Reply:
I'm impementing it in the https://github.com/joachimBurket/kodi_plugin_video_srgssr_ch_replay/tree/feature/main_menu branch. I'll merge it as soon as it is running